docs/how-to-build-with-ddn/with-mongodb.mdx

```diff
@@ -0,0 +1,399 @@
+---
+sidebar_position: 4
+sidebar_label: With MongoDB
+description: "Learn the basics of Hasura DDN and how to get started with a MongoDB database."
+keywords:
+  - hasura ddn
+  - graphql api
+  - getting started
+  - guide
+  - mongodb
+  - mongo
+---
+
+import Prereqs from "@site/docs/_prereqs.mdx";
+
+# Get Started with Hasura DDN and MongoDB
+
+## Overview
+
+This tutorial takes about fifteen minutes to complete. You'll learn how to:
+
+- Set up a new Hasura DDN project
+- Connect it to a MongoDB database
+- Generate Hasura metadata
+- Create a build
+- Run your first query
+
+Additionally, we'll familiarize you with the steps and workflows necessary to iterate on your API.
+
+This tutorial assumes you're starting from scratch. We'll use a locally running MongoDB instance via Docker and connect
+it to Hasura, but you can easily follow the steps if you already have data seeded; Hasura will never modify your source
+schema.
+
+<Prereqs />
+
+:::info Additionally, you'll need mongosh
+
+This tutorial uses mongosh — the Mongo shell — to interact with the local MongoDB instance. You can download it
+[here](https://www.mongodb.com/try/download/shell).
+
+:::
+
+## Tutorial
+
+### Step 1. Authenticate your CLI
+
+```sh title="Before you can create a new Hasura DDN project, you need to authenticate your CLI:"
+ddn auth login
+```
+
+This will launch a browser window prompting you to log in or sign up for Hasura DDN. After you log in, the CLI will
+acknowledge your login, giving you access to Hasura Cloud resources.
+
+### Step 2. Scaffold out a new local project
+
+```sh title="Next, create a new local project:"
+ddn supergraph init my-project && cd my-project
+```
+
+Once you move into this directory, you'll see your project scaffolded out for you. You can view the structure by either
+running `ls` in your terminal, or by opening the directory in your preferred editor.
+
+### Step 3. Initialize your MongoDB connector
+
+```sh title="In your project directory, run:"
+ddn connector init my_mongo -i
+```
+
+From the dropdown, start typing `mongo` and hit enter to accept the default port. Then, provide the following connection
+string:
+
+```plaintext
+mongodb://local.hasura.dev:27017/my_database
+```
+
+### Step 4. Start the local MongoDB container
+
+```sh title="Begin by creating a compose file for the Mongo service:"
+touch app/connector/my_mongo/compose.mongo.yaml
+```
+
+```yaml title="Then, open the file and add the following:"
+services:
+  mongodb:
+    image: mongo:latest
+    container_name: mongodb
+    ports:
+      - "27017:27017"
+```
+
+```sh title="Run the container:"
+docker compose -f app/connector/my_mongo/compose.mongo.yaml up -d
+```
+
+```sh title="Use the mongosh shell to seed the database:"
+docker exec -it mongodb mongosh --eval "
+db = db.getSiblingDB('my_database');
+db.users.insertMany([
+  { user_id: 1, name: 'Alice', age: 25 },
+  { user_id: 2, name: 'Bob', age: 30 },
+  { user_id: 3, name: 'Charlie', age: 35 }
+]);
+"
```

Comment:
Another way to select the database is to give it as an argument to the mongosh command:

```suggestion
docker exec -it mongodb mongosh my_database --eval "
db.users.insertMany([
  { user_id: 1, name: 'Alice', age: 25 },
  { user_id: 2, name: 'Bob', age: 30 },
  { user_id: 3, name: 'Charlie', age: 35 }
]);
"
```

It's a little more concise, although possibly it's easier to overlook the important database argument.